core/domain/services/graphql/mutators/BulkEntityMutator.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace EventEspresso\core\domain\services\graphql\mutators;
+
+use Exception;
+use OutOfBoundsException;
+use GraphQL\Type\Definition\ResolveInfo;
+use WPGraphQL\AppContext;
+
+class BulkEntityMutator extends EntityMutator
+{
+
+    /**
+     * @var callable $entity_mutator .
+     */
+    protected $entity_mutator;
+
+    /**
+     * BulkEntityMutator constructor.
+     *
+     * @param mixed $model  The model instance.
+     * @param array $entity_mutator The mutator for an entity.
+     */
+    public function __construct(callable $entity_mutator)
+    {
+        $this->entity_mutator = $entity_mutator;
+    }
+    /**
+     * Bulk updates entities.
+     *
+     * @param array       $input   The input for the mutation
+     * @param AppContext  $context The AppContext passed down to all resolvers
+     * @param ResolveInfo $info    The ResolveInfo passed down to all resolvers
+     * @return array
+     */
+    public function updateEntities($input, AppContext $context, ResolveInfo $info)
+    {
+
+        $updated = [];
+        $failed = [];
+        // TODO Add meaningful error messages for every failure.
+        // $errors = [];
+
+        try {
+            if (empty($input['uniqueInputs']) || !is_array($input['uniqueInputs'])) {
+                throw new OutOfBoundsException(
+                    esc_html__('A valid input was not provided.', 'event_espresso')
+                );
+            }
+
+            $sharedInput = ! empty($input['sharedInput']) ? $input['sharedInput'] : [];
+
+            foreach ($input['uniqueInputs'] as $uniqueInput) {
+                try {
+                    // values in $uniqueInput will override those in $sharedInput
+                    $finalInput = array_merge($sharedInput, $uniqueInput);
+                    // mutate the individual entity.
+                    call_user_func($this->entity_mutator, $finalInput, $context, $info);
+                    // we are here it means the update was successful.
+                    $updated[] = $uniqueInput['id'];
+                } catch (Exception $e) {
+                    // sorry mate, couldn't help you :(
+                    $failed[] = $uniqueInput['id'];
+                }
+            }
```

Comment:
if possible, it's best to avoid using `array_merge()` inside a foreach loop, here's some info:

- https://dev.to/klnjmm/never-use-arraymerge-in-a-for-loop-in-php-5go1
- https://github.com/dseguy/clearPHP/blob/master/rules/no-array_merge-in-loop.md
- https://github.com/kalessil/phpinspectionsea/issues/138

[EDIT] 
realized after writing the above that that won't work in this case because we need to pass `finalInput` to the mutator, but the above info is still nice to know.

one other thing though is that it's also better to just call functions dynamically instead of using `call_user_func()`,
again if possible, as there are always some edge cases where that sometimes doesn't work (like static methods)


so you might want to try something like the following for the above code:

```php
// mutate the individual entity.
{$this->entity_mutator}($finalInput, $context, $info);
```

if `{$this->entity_mutator}` doesn't work, you can first set that as a variable and then call that instead, ie:

```php
$mutator = $this->entity_mutator
$mutator($finalInput, $context, $info);
```

[ANOTHER EDIT] 
just scrolled down more and saw that it looks like the entity mutator methods ARE static, so you might have to continues to use `call_user_func()`, although maybe not since you are already setting those to the `$this->entity_mutator` property

Reply:
Thanks, I have used this

```php
$mutator = $this->entity_mutator
$mutator($finalInput, $context, $info);
```